src/connector/src/sink/iceberg/mod.rs

```diff
@@ -1315,14 +1415,77 @@ impl IcebergSinkCommitter {
 
 #[async_trait::async_trait]
 impl SinkCommitCoordinator for IcebergSinkCommitter {
-    async fn init(&mut self) -> Result<()> {
+    async fn init(&mut self) -> Result<Option<u64>> {
+        if self.is_exactly_once {
+            tracing::info!("Iceberg sink coordinator initing");
+            if self
+                .iceberg_sink_has_pre_commit_metadata(&self.db, self.param.sink_id.sink_id())
+                .await?
+            {
+                let ordered_metadata_list_by_end_epoch = self
+                    .get_pre_commit_info_by_sink_id(&self.db, self.param.sink_id.sink_id())
+                    .await?;
+
+                let mut last_recommit_epoch = 0;
+                for (end_epoch, sealized_bytes, snapshot_id, committed) in
+                    ordered_metadata_list_by_end_epoch
+                {
+                    let write_results_bytes = deserialize_metadata(sealized_bytes);
+                    let mut write_results = vec![];
+
+                    for each in write_results_bytes {
+                        let write_result = IcebergCommitResult::try_from_sealized_bytes(each)?;
+                        write_results.push(write_result);
+                    }
+
+                    match (
+                        committed,
+                        self.is_snapshot_id_in_iceberg(&self.config, snapshot_id)
+                            .await?,
+                    ) {
+                        (true, _) => {
+                            tracing::info!(
+                                "All data in log store has been written into external sink, skip"
```

Comment:
For all these logs, shall we add a sink_id to distinguish different sinks? Or should we change it to debug mode.

Reply:
Let's add `sink_id` first, and change it to debug mode after fully tested under the new framework.